### PR TITLE
[Enhancement] active mv automatically

### PR DESCRIPTION
**Pull request by murphyatwork**
Fixes #issue

Introduce a daemon thread `MVActiveChecker`:
1. Try to activate MV, if it's not set to inactive manually
2. The default check interval is 1 minute, user could modify it through `admin set frontend config('mv_active_checker_interval_seconds'='300')`

Use case:
1. The base table of MV could be dropped and created everyday, for Hive external table
2. Without this mechanism, user need to activate it in the workflow, which is tedious

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 40 / 48 (83.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|0|1|00.00%|[299]|
|:large_blue_circle:|com/starrocks/scheduler/MVActiveChecker.java|36|43|83.72%|[53, 54, 87, 88, 104, 105, 106]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 41 / 48 (85.42%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/MVActiveChecker.java|36|43|83.72%|[53, 54, 87, 88, 104, 105, 106]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

**Review by satanson — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/scheduler/MVActiveChecker.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+    @VisibleForTesting
+    public void runForTest() {
+        process();
+    }
+
+    private void process() {
+        Collection<Database> dbs = GlobalStateMgr.getCurrentState().getIdToDb().values();
+        for (Database db : CollectionUtils.emptyIfNull(dbs)) {
+            for (Table table : CollectionUtils.emptyIfNull(db.getTables())) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/scheduler/MVActiveChecker.java:66]: what will happen if dropping the db during processing.

**Comment by ABingHuang** [comment]
can we add a logic to active mv during mv refresh?

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 44 / 51 (86.27%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/scheduler/MVActiveChecker.java|36|43|83.72%|[53, 54, 87, 88, 104, 105, 106]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|3|3|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/scheduler/MVActiveChecker.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/scheduler/MVActiveChecker.java:66]: 1. the table list is got from a `ConcurrentHashMap`
2. `tryToActivate` will verify the existence of database again, throw exception if database doesn't exist

**Review by LiShuMing — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**